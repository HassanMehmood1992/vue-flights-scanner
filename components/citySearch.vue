<template>
  <v-autocomplete
    v-model="val"
    name="airportSearch"
    hide-no-data
    item-text="PlaceName"
    item-value="PlaceId"
    :class="classNames"
    :clearable="clearable"
    :color="color"
    :dark="dark"
    :deletable-chips="multiple"
    :flat="flat"
    :hide-details="hideDetails"
    :hint="hint"
    :items="items"
    :loading="loading"
    :multiple="multiple"
    :chips="multiple"
    :small-chips="multiple"
    :label="label"
    hide-selected
    :dense="dense"
    no-filter
    :outlined="outlined"
    :placeholder="placeholder"
    :prepend-inner-icon="icon"
    :return-object="returnObject"
    :required="required"
    :rounded="rounded"
    :rules="rules"
    :search-input.sync="searchInput"
    :solo="solo"
  >
    <template v-slot:label>
      <span>{{ label }}</span>
      <span v-if="required" class="red--text">*</span>
    </template>

    <!-- <template v-slot:item="data">
      <v-layout column>
        <div class="caption primary--text font-weight-medium">
          {{ data.item.icao ? data.item.icao : ""}}
          {{ data.item.icao && (data.item.iata || data.item.faaCode) ? " / " : "" }}
          {{ data.item.iata ? data.item.iata : data.item.faaCode ? data.item.faaCode : "" }}
        </div>
        <div class="caption grey--text">{{ data.item.cityCountryName}}</div>
      </v-layout>
    </template> -->
  </v-autocomplete>
</template>

<script>
import * as _ from "lodash";
export default {
  props: {
    classNames: {
      type: String,
      default: null,
    },
    clearable: {
      type: Boolean,
      default: false,
    },
    color: {
      type: String,
      default: null,
    },
    dark: {
      type: Boolean,
      default: false,
    },
    deletablechips: {
      type: Boolean,
      default: false,
    },
    dense: {
      type: Boolean,
      default: false,
    },
    flat: {
      type: Boolean,
      default: false,
    },
    forFilter: {
      type: Boolean,
      default: false,
    },
    hideDetails: {
      type: Boolean,
      default: false,
    },
    hint: {
      type: String,
      default: null,
    },
    label: {
      type: String,
      default: null,
    },
    multiple: {
      type: Boolean,
      default: false,
    },
    outlined: {
      type: Boolean,
      default: false,
    },
    placeholder: {
      type: String,
      default: null,
    },
    icon: {
      type: String,
      default: null,
    },
    returnObject: {
      type: Boolean,
      default: false,
    },
    required: {
      type: Boolean,
      default: false,
    },
    rounded: {
      type: Boolean,
      default: false,
    },
    rules: {
      type: Array,
      default: function () {
        return [];
      },
    },
    solo: {
      type: Boolean,
      default: false,
    },
    value: {
      type: [String, Object, Array],
      default: null,
    },

  },
  data: () => ({
    items: [],
    loading: false,
    searchInput: "",
  }),
  mounted() {},

  computed: {
    val: {
      get() {
        return this.value;
      },
      set(v) {
        if (_.isUndefined(v)) v = null;
        this.searchInput = "";
        this.$emit("onSelect", v);
      },
    },
  },

  watch: {
    // setAirports: {
    //   immediate: true,
    //   handler() {
    //     if (this.setAirports) {
    //       // if (!this.multiple) this.getAirportById(this.setAirports);
    //       // else this.getAirportByIds(this.setAirports);
    //       if (!this.multiple) {
    //         this.items = [this.setAirports];
    //       } else {
    //         this.items = this.setAirports;
    //       }

    //       //this.val = this.setAirports;
    //     }
    //   },
    // },
    searchInput: {
      immediate: false,
      handler() {
        if (this.searchInput) {
          this.getItems(this.searchInput);
        }
      },
    },
  },

  methods: {
    getItems: _.debounce(function (val) {
      if (val && val.length >= 2) {
        this.loading = true;
        axios
          .get(`autosuggest/v1.0/AE/AED/en-US/?query=${val}`)
          .then((response) => {
              this.items = response.data.Places
          })
          .finally(() => {
            this.loading = false;
          });
      }
    }, 1000),
 
  },
};
</script>