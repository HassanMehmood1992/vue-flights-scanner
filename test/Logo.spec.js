import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

import citySearch from '@/components/citySearch'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper

beforeEach(() => {
    let vuetify = new Vuetify()

    wrapper = mount(citySearch, {
        store: new Vuex.Store({
            state: { products: [] }
        }),
        localVue,
        vuetify,
        propsData: { label: 'Foobar' },
    })
})

afterEach(() => {
    wrapper.destroy()
})

describe('Header', () => {
  
    test('is fully functional', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
    test('should have a custom title and match snapshot', () => {

      // With jest we can create snapshot files of the HTML output
      // expect(wrapper.html()).toMatchSnapshot()
  
      // We could also verify this differently
      // by checking the text content
      
      debugger
      const title = wrapper.find('.v-select__slot > label > span')
      
      expect(title.element.innerHTML).toBe('Foobar')
    })
})