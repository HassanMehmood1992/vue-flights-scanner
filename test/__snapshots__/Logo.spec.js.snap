// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is fully functional 1`] = `
<div
  class="v-input theme--light v-text-field v-select v-autocomplete"
>
  <div
    class="v-input__control"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-owns="list-4"
      class="v-input__slot"
      role="combobox"
    >
      <div
        class="v-select__slot"
      >
        <label
          class="v-label theme--light"
          for="input-4"
          style="left: 0px; position: absolute;"
        >
          <span>
            Foobar
          </span>
           
          <!---->
        </label>
        <input
          autocomplete="off"
          id="input-4"
          name="airportSearch"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-menu-down theme--light"
            />
          </div>
        </div>
        <input
          name="airportSearch"
          type="hidden"
        />
      </div>
      <div
        class="v-menu"
      >
        <!---->
      </div>
    </div>
    <div
      class="v-text-field__details"
    >
      <div
        class="v-messages theme--light"
      >
        <transition-group-stub
          class="v-messages__wrapper"
          name="message-transition"
          tag="div"
        />
      </div>
    </div>
  </div>
</div>
`;
